migrations/2024-02-12-211114_add_vote_display_mode_setting/up.sql

```diff
@@ -0,0 +1,20 @@
+-- Create an extra table to hold local user vote display settings
+-- Score and Upvote percentage are turned on by default.
+CREATE TABLE local_user_vote_display_mode (
+    local_user_id int REFERENCES local_user ON UPDATE CASCADE ON DELETE CASCADE NOT NULL,
+    score boolean DEFAULT TRUE NOT NULL,
+    upvotes boolean DEFAULT FALSE NOT NULL,
+    downvotes boolean DEFAULT FALSE NOT NULL,
+    upvote_percentage boolean DEFAULT TRUE NOT NULL,
+    published timestamp with time zone NOT NULL DEFAULT now(),
+    updated timestamp with time zone,
+    PRIMARY KEY (local_user_id)
```

Comment:
Why not add this directly to the `local_user` table? Also whats the reason for having published and updated columns? Not even `local_user` has those and they dont seem to serve any purpose.

Reply:
Logically, they should, but diesel has those column limits that drastically increase compile time, that you can only increase via flags. https://docs.diesel.rs/master/diesel/index.html

We're already almost at 32 columns, and we had to do a similar thing to `LocalSite` with breaking out `LocalSiteRateLimit` into its own table.

Reply:
Its a join, so it needs to be included in `LocalUserView` like in this PR. It does need to be well typed as its own table.

We used to try to do things like this with views, but overall its not worth it. This works the same way as `LocalSiteView`, with `local_site_rate_limit` as a joined table.

Reply:
I think you can do something like this:

```
struct LocalUser{}
struct LocalUser2{}
struct LocalUserWrapper{
    #[serde(flatten)
    local_user_1: LocalUser, 
    #[serde(flatten)
    local_user_2: LocalUser2
}
struct LocalUserView {
    local_user: LocalUserWrapper,
    ...
}
```

Then the api should return all of it directly in `LocalUserView.local_user`

Reply:
Almost forgot, you also need to ensure this is included in the settings export/import.

https://github.com/LemmyNet/lemmy/blob/main/crates/apub/src/api/user_settings_backup.rs

Reply:
I'd really rather not do this, it goes against the standard previously set by LocalSiteRateLimit joined to LocalSite, and might have some major issues with lemmy-ui.

Reply:
The UserSettings backup changes are included in this PR.

Reply:
Alright but I would still use a more generic name like `local_user_extra` because we will definitely add more settings over time. Would be annoying if we have to rename everything then, or add another table.

Reply:
There will likely be other tables, I plan to add `local_user_notification` as a separate table as well, so `local_user_extra` wouldn't be a good name for this.

Reply:
Alright. This can be merged then?